### PR TITLE
Fix cmake build on linux, OpenCL link dependencies missing from C lib…

### PR DESCRIPTION
**Pull request by cianx**
…rary.

Signed-off-by: Cian Montgomery <cianmontgomery@gmail.com>

### CODE REVIEW

**Comment by cianx** [comment]
Prior to this change make was throwing the following errors:

```~/Workspaces/xmr-stak-amd$ make
Scanning dependencies of target xmr-stak-c
[  5%] Building C object CMakeFiles/xmr-stak-c.dir/amd_gpu/gpu.c.o
[ 10%] Linking C static library libxmr-stak-c.a
[ 42%] Built target xmr-stak-c
Scanning dependencies of target xmr-stak-amd
[ 47%] Building CXX object CMakeFiles/xmr-stak-amd.dir/cli-miner.cpp.o
[ 52%] Building CXX object CMakeFiles/xmr-stak-amd.dir/executor.cpp.o
[ 57%] Linking CXX executable bin/xmr-stak-amd
libxmr-stak-c.a(gpu.c.o): In function `InitOpenCLGpu':
gpu.c:(.text+0x110): undefined reference to `clGetDeviceInfo'
gpu.c:(.text+0x1ad): undefined reference to `clCreateCommandQueueWithProperties'
gpu.c:(.text+0x205): undefined reference to `clCreateBuffer'
gpu.c:(.text+0x25b): undefined reference to `clCreateBuffer'
gpu.c:(.text+0x2ad): undefined reference to `clCreateBuffer'
gpu.c:(.text+0x308): undefined reference to `clCreateBuffer'
gpu.c:(.text+0x356): undefined reference to `clCreateBuffer'
libxmr-stak-c.a(gpu.c.o):gpu.c:(.text+0x3a4): more undefined references to `clCreateBuffer' follow
libxmr-stak-c.a(gpu.c.o): In function `InitOpenCLGpu':
gpu.c:(.text+0x492): undefined reference to `clCreateProgramWithSource'
gpu.c:(.text+0x511): undefined reference to `clBuildProgram'
gpu.c:(.text+0x55a): undefined reference to `clGetProgramBuildInfo'
gpu.c:(.text+0x5bd): undefined reference to `clGetProgramBuildInfo'
gpu.c:(.text+0x630): undefined reference to `clGetProgramBuildInfo'
gpu.c:(.text+0x6a7): undefined reference to `clCreateKernel'
libxmr-stak-c.a(gpu.c.o): In function `InitOpenCL':
gpu.c:(.text+0x762): undefined reference to `clGetPlatformIDs'
gpu.c:(.text+0x7dc): undefined reference to `clGetPlatformIDs'
gpu.c:(.text+0x845): undefined reference to `clGetDeviceIDs'
gpu.c:(.text+0x8f0): undefined reference to `clGetDeviceIDs'
gpu.c:(.text+0x999): undefined reference to `clCreateContext'
libxmr-stak-c.a(gpu.c.o): In function `XMRSetJob':
gpu.c:(.text+0xaec): undefined reference to `clEnqueueWriteBuffer'
gpu.c:(.text+0xb70): undefined reference to `clSetKernelArg'
gpu.c:(.text+0xbd6): undefined reference to `clSetKernelArg'
gpu.c:(.text+0xc2d): undefined reference to `clSetKernelArg'
gpu.c:(.text+0xc6f): undefined reference to `clSetKernelArg'
gpu.c:(.text+0xcb4): undefined reference to `clSetKernelArg'
libxmr-stak-c.a(gpu.c.o):gpu.c:(.text+0xcf9): more undefined references to `clSetKernelArg' follow
libxmr-stak-c.a(gpu.c.o): In function `XMRRunJob':
gpu.c:(.text+0x1046): undefined reference to `clEnqueueWriteBuffer'
gpu.c:(.text+0x1097): undefined reference to `clEnqueueWriteBuffer'
gpu.c:(.text+0x1136): undefined reference to `clFinish'
gpu.c:(.text+0x119f): undefined reference to `clEnqueueNDRangeKernel'
gpu.c:(.text+0x1207): undefined reference to `clEnqueueNDRangeKernel'
gpu.c:(.text+0x126f): undefined reference to `clEnqueueNDRangeKernel'
gpu.c:(.text+0x12dd): undefined reference to `clEnqueueReadBuffer'
gpu.c:(.text+0x131d): undefined reference to `clEnqueueReadBuffer'
gpu.c:(.text+0x135d): undefined reference to `clEnqueueReadBuffer'
gpu.c:(.text+0x139d): undefined reference to `clEnqueueReadBuffer'
gpu.c:(.text+0x13c5): undefined reference to `clFinish'
gpu.c:(.text+0x1401): undefined reference to `clEnqueueReadBuffer'
gpu.c:(.text+0x1417): undefined reference to `clFinish'
gpu.c:(.text+0x1445): undefined reference to `clSetKernelArg'
gpu.c:(.text+0x14c6): undefined reference to `clEnqueueNDRangeKernel'
collect2: error: ld returned 1 exit status
CMakeFiles/xmr-stak-amd.dir/build.make:333: recipe for target 'bin/xmr-stak-amd' failed
make[2]: *** [bin/xmr-stak-amd] Error 1
CMakeFiles/Makefile2:104: recipe for target 'CMakeFiles/xmr-stak-amd.dir/all' failed
make[1]: *** [CMakeFiles/xmr-stak-amd.dir/all] Error 2
Makefile:127: recipe for target 'all' failed
make: *** [all] Error 2
```

File: CMakeLists.txt [hunk 1]
```diff
@@ -109,18 +109,18 @@ endif()
 file(GLOB SRCFILES_CPP "*.cpp" "crypto/*.cpp")
 file(GLOB SRCFILES_C "crypto/*.c" "amd_gpu/*.c")
 
-add_library(xmr-stak-c
+add_library(xmr-stak-amd-c
```
Comment (psychocrypt) [inline CMakeLists.txt]: The renaming is not necessary but from my site ok.

**Comment by psychocrypt** [comment]
Could you please post your compiler and OS.

**Comment by cianx** [comment]
$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 16.04.2 LTS
Release:	16.04
Codename:	xenial
$ c++ --version
c++ (Ubuntu 5.4.0-6ubuntu1~16.04.4) 5.4.0 20160609
Copyright (C) 2015 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

$ cmake --version
cmake version 3.5.1

CMake suite maintained and supported by Kitware (kitware.com/cmake).

**Comment by psychocrypt** [comment]
thx @cianx  for reporting and fixing

@fireice-uk I can reproduce the issue and we can merge this fix.

**Comment by psychocrypt** [comment]
@fireice-uk  Could you please merge this bug fix

**Comment by fireice-uk** [comment]
Ok. Done.

**Pull request merged**